gnovm/pkg/doc/dirs.go

```diff
@@ -0,0 +1,163 @@
+// Mostly copied from go source at tip, commit d922c0a.
+//
+// Copyright 2015 The Go Authors. All rights reserved.
+
+package doc
+
+import (
+	"log"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+)
+
+// A Dir describes a directory holding code by specifying
+// the expected import path and the file system directory.
+type Dir struct {
```

Comment:
Could all this Dir logic be replaced by https://pkg.go.dev/path/filepath#WalkDir ?